### ISSUE TITLE
Notification of Tx immediately without consensus

### ISSUE THREAD

**Issue by buhrmi** [babbleio/babble#5]
In the Swirlds Platform, the `handleTransaction` method (equivalent to the Babble `commitTx` method) has the signature `(long id, boolean consensus, time timestamp, byte[] payload)` ... So the platform notifies the app of a transaction even though the "consensus timestamp" hasn't been calculated yet. Is it possible to add this in Babble?

**Comment by arrivets** [comment]
Yeah it is definitely possible to add this. Good idea. @mpitid is working on a new interface between client and node so we will try to add it as part of that pull request.

**Comment by buhrmi** [comment]
That's cool, is there something to look at? For my stock exchange, I'd need to connect to it as tightly as possible and probably mirror the interface using websockets

**Comment by arrivets** [comment]
The pull request is here: https://github.com/babbleio/babble/pull/3
It is still work in progress.
The client subscribes to a channel, stays connected and receives commitTx messages from the node. Similar to websockets.

**Comment by buhrmi** [comment]
Ah, I prefer to not have the concept of individual channels inside the core tech. it should be a global bus and the question of whether or not a message belongs into a given channel should be up for the app developer to decide. so, imho, the concept of channels should not appear in the go source code

**Comment by arrivets** [comment]
My bad, I didnt explain it well. There is only a single stream of transactions that the client can subscribe to. The client receives ALL transactions, not just the ones pertaining to a specific channel. So yeah the work of slitting transactions should be done outside of Babble.

**Comment by buhrmi** [comment]
ah okay. makes sense.
was looking at the PR. looks like it implements an HTTP interface while the old interface was a lower level TCP interface. correct? Would be good if the raw TCP interface could include the functionality too because I think the nodejs process should connect to babble via TCP, not HTTP

**Comment by arrivets** [comment]
The PR introduces an API abstraction. At the moment, there is only an HTTP implementation but it is possible to implement the API with lower level TCP. In fact I think @mpitid has already written one but it is not committed yet. Just as a matter of curiosity, why do you prefer to use raw TCP between the nodejs process and babble? Is it a matter of performance? Anyway there is definitely going to be a raw TCP implementation of the API

**Comment by buhrmi** [comment]
Yeah, for performance.
